batch/batch/instance_config.py

```diff
@@ -106,13 +110,19 @@ def cost_per_hour_from_cores(
         utilized_cores_mcpu: int,
     ) -> float:
         assert 0 <= utilized_cores_mcpu <= self.cores * 1000
-        memory_in_bytes = self.cores_mcpu_to_memory_bytes(utilized_cores_mcpu)
+        if self.job_private:
+            memory_in_bytes = self.instance_memory()
+        else:
+            memory_in_bytes = self.cores_mcpu_to_memory_bytes(utilized_cores_mcpu)
         storage_in_gb = 0  # we don't need to account for external storage
         return self.cost_per_hour(resource_rates, utilized_cores_mcpu, memory_in_bytes, storage_in_gb)
 
     def actual_cost_per_hour(self, resource_rates: Dict[str, float]) -> float:
         cpu_in_mcpu = self.cores * 1000
-        memory_in_bytes = self.cores_mcpu_to_memory_bytes(cpu_in_mcpu)
+        if self.job_private:
+            memory_in_bytes = self.instance_memory()
+        else:
+            memory_in_bytes = self.cores_mcpu_to_memory_bytes(cpu_in_mcpu)
```

Comment:
Can we not always use `self.instance_memory()` here? I'm interpreting this as the cost per hour of the whole instance, since we have `cpu_in_mcpu = self.cores * 1000`.